### PR TITLE
AEA-961 CLI register command updated due to opt-in marketing requirements.

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

CLI register command updated due to opt-in marketing requirements.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/register.py [hunk 1]
```diff
@@ -55,4 +55,6 @@ def do_register(
     username = validate_author_name(username)
     token = register_new_account(username, email, password, password_confirmation)
     update_cli_config({AUTH_TOKEN_KEY: token})
+    if click.confirm('Do you want to subscribe?'):
+        click.echo("Please visit aea-registry.fetch.ai/mailing-list to subscribe")
```
Comment (DavidMinarsch) [inline aea/cli/register.py:59]:
```suggestion
        click.echo("Please visit `https://aea-registry.fetch.ai/mailing-list` to subscribe for developer news.")
```

File: aea/cli/register.py [hunk 2]
```diff
@@ -55,4 +55,6 @@ def do_register(
     username = validate_author_name(username)
     token = register_new_account(username, email, password, password_confirmation)
     update_cli_config({AUTH_TOKEN_KEY: token})
+    if click.confirm('Do you want to subscribe?'):
```
Comment (DavidMinarsch) [inline aea/cli/register.py:58]:
```suggestion
    if click.confirm('Do you want to subscribe for developer news?'):
```

Comment (DavidMinarsch) [inline aea/cli/register.py:58]: There need's to be a flag provided so you can override the prompt via the flag; the flag should be propagated to `init` as well

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
LGTM

File: aea/cli/register.py [hunk 3]
```diff
@@ -49,10 +56,16 @@ def do_register(
...
     :return: None
     """
     username = validate_author_name(username)
     token = register_new_account(username, email, password, password_confirmation)
     update_cli_config({AUTH_TOKEN_KEY: token})
+    if subscribe or click.confirm("Do you want to subscribe for developer news?"):
+        click.echo(
+            "Please visit `https://aea-registry.fetch.ai/mailing-list` "
+            "to subscribe for developer news"
+        )
```
Comment (marcofavorito) [inline aea/cli/register.py:70]: so the only effect of `--subscribe` is to print the URL to the command line?

Comment (panasevychol) [reply aea/cli/register.py:70]: Yes. That was the requirement.

**Pull request merged**